mlrun/runtimes/function.py

```diff
@@ -969,6 +975,11 @@ def compile_function_config(function: RemoteRuntime):
         spec.set_config("spec.nodeName", function.spec.node_name)
     if function.spec.affinity:
         spec.set_config("spec.affinity", function.spec._get_sanitized_affinity())
+    # don't send default or any priority class name if nuclio is not compatible
+    if function.spec.priority_class_name and validate_nuclio_version_compatibility(
+        "1.6.18"
+    ):
```

Comment:
doing the validation both on server and client side sounds right, you can do the same to the function node selection options (in this or a separate PR)